pyrate/core/dem_error.py

```diff
@@ -0,0 +1,307 @@
+
+    # save angles to geotiff
+    gt, md, wkt = shared.get_geotiff_header_info(ifg_path)
+    md[ifc.EPOCH_DATE] = None # needs to have a value in write_output_geotiff
+    look_angle_file = os.path.join(params[cf.OUT_DIR], 'look_angle.tif')
+    shared.write_output_geotiff(md, gt, wkt, look_angle, look_angle_file, np.nan)
+    incidence_angle_file = os.path.join(params[cf.OUT_DIR], 'incidence_angle.tif')
+    shared.write_output_geotiff(md, gt, wkt, incidence_angle, incidence_angle_file, np.nan)
+    azimuth_angle_file = os.path.join(params[cf.OUT_DIR], 'azimuth_angle.tif')
+    shared.write_output_geotiff(md, gt, wkt, azimuth_angle, azimuth_angle_file, np.nan)
+    incidence_angle_gamma_file = os.path.join(params[cf.OUT_DIR], 'incidence_angle_gamma.tif')
+    shared.write_output_geotiff(md, gt, wkt, incidence_angle_gamma, incidence_angle_gamma_file, np.nan)
+    azimuth_angle_gamma_file = os.path.join(params[cf.OUT_DIR], 'azimuth_angle_gamma.tif')
+    shared.write_output_geotiff(md, gt, wkt, azimuth_angle_gamma, azimuth_angle_gamma_file, np.nan)
+
+    return look_angle
+
+
+def calc_local_baseline(ifg, az, look_angle, params):
+    """
+    Function to calculate perpendicular baseline values for each pixel.
+    """
+    # open ifg object
+    if not ifg.is_open:
+        ifg.open()
+
+    # read relevant metadata from IFG
+    prf = float(ifg.meta_data[ifc.PYRATE_PRF_HERTZ])
+    az_looks = int(ifg.meta_data[ifc.PYRATE_AZIMUTH_LOOKS])
+    az_n = int(ifg.meta_data[ifc.PYRATE_AZIMUTH_N])
+    #base_T = float(ifg.meta_data[ifc.PYRATE_BASELINE_T])
+    base_C = float(ifg.meta_data[ifc.PYRATE_BASELINE_C])
+    base_N = float(ifg.meta_data[ifc.PYRATE_BASELINE_N])
+    #baserate_T = float(ifg.meta_data[ifc.PYRATE_BASELINE_RATE_T])
+    baserate_C = float(ifg.meta_data[ifc.PYRATE_BASELINE_RATE_C])
+    baserate_N = float(ifg.meta_data[ifc.PYRATE_BASELINE_RATE_N])
+
+    # calculate per pixel baseline vectors across track (C) and normal to the track (N)
+    mean_az = az_n / 2 - 0.5 # mean azimuth line
+    prf = prf / az_looks # Pulse Repetition Frequency needs to be adjusted according to GAMMA azimuth looks
+    base_C_local = base_C + baserate_C * (az - mean_az) / prf
+    base_N_local = base_N + baserate_N * (az - mean_az) / prf
+
+    # calculate the per-pixel perpendicular baseline
+    bperp = np.multiply(base_C_local, np.cos(look_angle)) - np.multiply(base_N_local, np.sin(look_angle))
+
+    # save perpendicular baseline to npy array
+    bperp_on_disc = MultiplePaths.bperp_path(ifg.data_path, params)
+    np.save(file=bperp_on_disc, arr=bperp)
+
+    # save bperp to geotiff (temporary for visualisation)
+    gt, md, wkt = shared.get_geotiff_header_info(ifg.data_path)
+    md[ifc.EPOCH_DATE] = None # needs to have a value in write_output_geotiff
+    bperp_tif_on_disc = MultiplePaths.bperp_tif_path(ifg.data_path, params)
+    shared.write_output_geotiff(md, gt, wkt, bperp, str(bperp_tif_on_disc), np.nan)
```

Comment:
As before, tif be default. npy only if `savenpy = 1`

Reply:
Deleted (was for testing only).